packages/sdk/CONTRIBUTING.md

```diff
@@ -0,0 +1,252 @@
+
+0-level Types can use Domain Types, not the other way around. 
+
+## 0-level Types
+
+In most cases, Voucherify SDK methods are just wrappers for HTTP clients, so 0-level Types should be aligned to the Voucherify OpenAPI file in the context of:
+- resource naming
+- types naming
+- data structure, property types, required/optional attribute
+
+
+Because of that, the 0-level Types naming should be aligned with [OpenAPI Types](https://github.com/voucherifyio/voucherify-openapi/blob/master/CONTRIBUTING.md#openapi), but instead of snake casing, we will use the pascal case casing used in SDK.
+
+Pattern: `<resource><action><request|response><body|param|query>`, where 
+	- `resource`: plural name taken from API path, e.g. `vouchers`, `customers`, `products` 
+	- `action` : `get`(single record), `list`, `update`, `delete`, `create` (etc.)
+	- `request` or `response`
+	- `body`, `param` or `query` 
+
+If in 0-level Types, there is a need to refer to created child types (enums, list items, other large structures), and those child types are used only for this one 0-level, we can still name them as 0-level type, including additional information after `action` and following the pattern:
+
+`<resource><action><child type distinction><request|response><body|param|query>`
+
+For example:
+
+The 0-level type that describes the voucher list response body is named `VouchersListResponseBody`, but the voucher object is large itself; it's also a good practice to give the SDK user the type of the specific list item so he can use it easily in his application, so we should create the type describing voucher object and name it: `VouchersListItemResponseBody`.
+
+## Domain Types
+
+Domain Types:
+- should not be identified with any specific API endpoint or SDK method
+- should prefer composition over inherence
+- names should describe what it's inside, not where it's used, e.g: `CustomerAddress`, `CustomerSummary`
+- should have optional fields; we can mark them as required in 0-level Types
+- avoid utility types like: Pick, Omit, Partial
+
+> [!NOTE] Do not create domain type if we can't give intuitive name
+
+## Examples
+> [!WARNING] This example shows how we can structure the types, it's not Voucherify specification
+
+```ts
+// Utility types 👇
+
+// Mark specific property as required
+type WithRequiredProperty<Type, Key extends keyof Type> = Type & {
+  [Property in Key]-?: Type[Property];
+};
+// Remove null as the option type for specific properties
+type WithNonNullableProperty<Type, Key extends keyof Type> = Type & {
+  [Property in Key]: Exclude<Type[Property], null> ;
+};
+
+// Mark all nested properties as required
+// Use DeepRequired<> from `utility-types` - https://www.npmjs.com/package/utility-types#deeprequiredt
+
+// Domain types 👇
+
+// Basic customer properties, common for most of the 0-level endpoints
+type CustomerBase = {
+	name?: string
+	email?: string
+	phone?: string
+	description?: string
+	birthdate?: string
+	metadata?: Record<string, any>
```

Comment:
Change "any" to "unknown" -> you still can assign here anything, but it does not turn off all TS safeguards when you read it